### PR TITLE
Remove EasyMock dependency from CalciteTests.

### PR DESCRIPTION
**Pull request by gianm**
Useful because CalciteTests is used by other modules (e.g. druid-benchmarks)
and we don't want them to have to pull in EasyMock.

### CODE REVIEW

File: sql/src/test/java/org/apache/druid/sql/calcite/util/CalciteTests.java [hunk 1]
```diff
@@ -129,20 +133,24 @@
 import org.apache.druid.sql.calcite.view.ViewManager;
 import org.apache.druid.timeline.DataSegment;
 import org.apache.druid.timeline.partition.LinearShardSpec;
-import org.easymock.EasyMock;
```
Comment (clintropolis) [inline sql/src/test/java/org/apache/druid/sql/calcite/util/CalciteTests.java]: Hmm, I've got an open PR that is using EasyMock heavily to allow for testing sql queries on system tables....

https://github.com/apache/druid/pull/6974/files#diff-ed8d85db2f346595b7faf34f45a78fd6R870

Any ideas on what we should do?

Comment (gianm) [reply sql/src/test/java/org/apache/druid/sql/calcite/util/CalciteTests.java]: Hmm, probably either modify FakeDruidNodeDiscoveryProvider to return actual servers (it's a simple thing to subclass, or restore EasyMock in druid-sql and find another way to solve the original problem (https://github.com/apache/druid/pull/9308#issuecomment-582128976) — a suppression or something else.

Comment (gianm) [reply sql/src/test/java/org/apache/druid/sql/calcite/util/CalciteTests.java]: If you want to do the former, we can merge this and then you can update your PR; if you want to do the latter, I can change this patch to add a suppression. I'm ok either way so let me know what you think is best.

(But check out the thread on dev@druid about EasyMock first 🙂)

Comment (clintropolis) [reply sql/src/test/java/org/apache/druid/sql/calcite/util/CalciteTests.java]: To do the former i would also need to implement `FakeDruidNodeDiscovery` too I think for the provider to provide to get the list of servers.. which i guess isn't that big a deal probably...

**Pull request merged**